framework/log/FileTarget.php

```diff
@@ -137,25 +142,70 @@ protected function rotateFiles()
         for ($i = $this->maxLogFiles; $i >= 0; --$i) {
             // $i == 0 is the original log file
             $rotateFile = $file . ($i === 0 ? '' : '.' . $i);
-            if (is_file($rotateFile)) {
-                // suppress errors because it's possible multiple processes enter into this section
-                if ($i === $this->maxLogFiles) {
-                    @unlink($rotateFile);
+            $rotateTarget = $file . '.' . ($i + 1);
+            if (!is_file($rotateFile)) {
```

Comment:
there is no else on this one :) only if the file does not exist, something needs to be done.